plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -700,7 +699,7 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
                 .map(JdbcColumnHandle.class::cast)
                 .collect(toImmutableList());
         JdbcOutputTableHandle handle = jdbcClient.beginInsertTable(session, (JdbcTableHandle) tableHandle, columnHandles);
-        if (!isNonTransactionalInsert(session)) {
+        if (!handle.getTableName().equals(handle.getTemporaryTableName())) {
```

Comment:
Thanks for improving the logic, but this doesn't change the fact the logic belongs to jdbcClient.rollbackCreateTable, as it is coupled with jdbcClient.beginInsertTable.

As suggested in https://github.com/trinodb/trino/pull/12229/commits/a1f6ada23ada49944c665f1307af1ff7aa0a48f6#r866674664, let's make temporaryTableName `Optional`.

Reply:
I pushed a new update. (Will rename squash/rename the commits later).

Not sure I like.
Perhaps I could abstract the repeated handle.getTemporaryTableName().orElse(handle.getTableName()) into JdbcOutputTableHanlde itself, and add a separate method to check whether a temporary table has been provided.